frontend/src/components/elements/PlotlyChart/PlotlyChart.tsx

```diff
@@ -30,58 +30,87 @@ export interface PlotlyChartProps {
   height: number | undefined
 }
 
+export interface PlotlyIFrameProps {
+  width: number
+  height: number | undefined
+  url: string
+}
+
 export const DEFAULT_HEIGHT = 450
 
-export function PlotlyChart({
-  width: propWidth,
-  element,
+function renderIFrame({
+  url,
+  width,
   height: propHeight,
+}: PlotlyIFrameProps): ReactElement {
+  const height = propHeight || DEFAULT_HEIGHT
+  return <iframe title="Plotly" src={url} style={{ width, height }} />
+}
+
+function renderFigure({
+  element,
+  width,
+  height,
 }: PlotlyChartProps): ReactElement {
-  const renderIFrame = (url: string): ReactElement => {
-    const height = propHeight || DEFAULT_HEIGHT
-    const width = propWidth
-    return <iframe title="Plotly" src={url} style={{ width, height }} />
-  }
+  const figure = element.figure as FigureProto
+  const isFullScreen = (): boolean => !!height
```

Comment:
nit: the fullscreen wrapper also directly provides an `isFullScreen` boolean property. Not sure if it makes any difference here, but it might be a bit more future-proof:

https://github.com/streamlit/streamlit/blob/7550b6e2edfdc2c3a11fd6b68d8e64541963237a/frontend/src/hocs/withFullScreenWrapper/withFullScreenWrapper.tsx#L47